src/Application/Migrations/17_function_ticket_dependency_satisfied.sql

```diff
@@ -2,39 +2,33 @@
 
 SET ROLE TO postgres;
 
-CREATE OR REPLACE FUNCTION ticket_depending_encoding_ticket_state_satisfied(param_ticket_id bigint)
+CREATE OR REPLACE FUNCTION ticket_dependee_ticket_state_satisfied(param_depender_ticket_id bigint)
   RETURNS boolean AS
   $$
 DECLARE
-	state enum_ticket_state;
 	satisfaction boolean;
+	dependee_state enum_ticket_state;
 BEGIN
+	SELECT ticket_dependee_ticket_state(param_depender_ticket_id)
+		INTO dependee_state;
+
 	SELECT
-		masterstate.sort >= wantedstate.sort INTO satisfaction
+		dependee_ticket_state.sort >= configured_trigger_state.sort INTO satisfaction
 	FROM
-		tbl_ticket t
-	JOIN
-		tbl_encoding_profile_version epv ON epv.id = t.encoding_profile_version_id
-	JOIN
-		tbl_encoding_profile ep ON epv.encoding_profile_id = ep.id
-	JOIN
-		tbl_encoding_profile ep2 ON ep2.id = ep.depends_on
-	JOIN
-		tbl_encoding_profile_version epv2 ON epv2.encoding_profile_id = ep2.id
-	JOIN
-		tbl_ticket t2 ON t2.encoding_profile_version_id = epv2.id AND t2.parent_id = t.parent_id
+		tbl_ticket depender_ticket
 	JOIN
-		tbl_project p ON t.project_id = p.id
+		tbl_project project ON param_ticket.project_id = project.id
 	JOIN
-		tbl_ticket_state masterstate ON
-			masterstate.ticket_type = t2.ticket_type AND
-			masterstate.ticket_state = t2.ticket_state
+		-- if given ticket is no encoding ticket, function will return NULL
+		tbl_ticket_state dependee_ticket_state ON
+			depender_ticket.ticket_type = 'encoding' AND
+			dependee_ticket_state.ticket_state = dependee_state
 	JOIN
-		tbl_ticket_state wantedstate ON
-			wantedstate.ticket_type = t2.ticket_type AND
-			wantedstate.ticket_state = p.dependent_ticket_trigger_state
+		tbl_ticket_state configured_trigger_state ON
+			depender_ticket.ticket_type = 'encoding' AND
```

Comment:
But ticket_state is an enum and therefore unique, isnt it? The JOIN is meant to deliver the state attributes for encoding tickets and NULLs for other ticket types. So is your concern that these JOINS produce more tupels? I cannot think of a case where this might happen.

If your concern is that dependee_ticket_state and configured_trigger_state might contain the same tupel of the state table, then yes this is possible of course.

Reply:
ticket_state is an enum, but tbl_ticket_state is a table (which you are joining here) and it contains multiple rows with e.g. ```ticket_state = 'removing'```. I know, it's not really likely that the trigger state would be one of those, but technically it could happen.

> The JOIN is meant to deliver the state attributes for **encoding** tickets
But the ticket_type of **tbl_ticket_state** is not part of any condition... That's what I'm referring to.

Reply:
Looks good, although I'd have used ```dependee_ticket_state.ticket_type = depender_ticket.ticket_type``` just to have the hard-coded value only once.

Reply:
And since you don't do any left joins and such, I'd suggest to move the ```depender_ticket.ticket_type = 'encoding'``` condition to the where clause, to only have it once. The function should still return NULL, if the ticket is not a encoding ticket.